create-publication-from-doi/src/test/java/no/unit/nva/doi/fetch/MainHandlerTest.java

```diff
@@ -335,8 +334,9 @@ private OutputStream outputStream() {
         return new ByteArrayOutputStream();
     }
 
-    private GatewayResponse gatewayResponse(OutputStream outputStream) throws JsonProcessingException {
-        return objectMapper.readValue(outputStream.toString(), GatewayResponse.class);
+    private GatewayResponse<String> gatewayResponse(OutputStream outputStream) throws JsonProcessingException {
```

Comment:
It is wrong to assume that you expect any kind of OutputStream here because you make the assumption that `outputStream.toString()` returns the contents of the stream. That it not true in the general case. Only in `ByteArrayOutputStream`.

Reply:
In any case this method is not necessary if you you the private methods suggested above and you use the Problem.getDetails()